test/common/event/file_event_impl_test.cc

```diff
@@ -439,6 +439,41 @@ TEST_F(FileEventImplTest, RegisterIfEmulatedEdge) {
   dispatcher_->run(Event::Dispatcher::RunType::NonBlock);
 }
 
+#ifndef WIN32 // This is a test case that clarifies the event behavior.
```

Comment:
I would think that this test would also work on Windows.

Reply:
It failed on windows CI. It's not too surprised to see the behavior is not cross platform